crates/router/src/connector/prophetpay.rs

```diff
@@ -0,0 +1,534 @@
+pub mod transformers;
+
+use std::fmt::Debug;
+
+use error_stack::{IntoReport, ResultExt};
+use masking::ExposeInterface;
+use transformers as prophetpay;
+
+use crate::{
+    configs::settings,
+    core::errors::{self, CustomResult},
+    headers,
+    services::{
+        self,
+        request::{self, Mask},
+        ConnectorIntegration, ConnectorValidation,
+    },
+    types::{
+        self,
+        api::{self, ConnectorCommon, ConnectorCommonExt},
+        ErrorResponse, Response,
+    },
+    utils::{self, BytesExt},
+};
+
+#[derive(Debug, Clone)]
+pub struct Prophetpay;
+impl api::payments::MandateSetup for Prophetpay {}
+impl api::Payment for Prophetpay {}
+impl api::PaymentSession for Prophetpay {}
+impl api::ConnectorAccessToken for Prophetpay {}
+impl api::PaymentAuthorize for Prophetpay {}
+impl api::PaymentSync for Prophetpay {}
+impl api::PaymentCapture for Prophetpay {}
+impl api::PaymentVoid for Prophetpay {}
+impl api::Refund for Prophetpay {}
+impl api::RefundExecute for Prophetpay {}
+impl api::RefundSync for Prophetpay {}
+impl api::PaymentToken for Prophetpay {}
+
+impl
+    ConnectorIntegration<
+        api::PaymentMethodToken,
+        types::PaymentMethodTokenizationData,
+        types::PaymentsResponseData,
+    > for Prophetpay
+{
+    // Not Implemented (R)
+}
+
+impl<Flow, Request, Response> ConnectorCommonExt<Flow, Request, Response> for Prophetpay
+where
+    Self: ConnectorIntegration<Flow, Request, Response>,
+{
+    fn build_headers(
+        &self,
+        req: &types::RouterData<Flow, Request, Response>,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, request::Maskable<String>)>, errors::ConnectorError> {
+        let mut header = vec![(
+            headers::CONTENT_TYPE.to_string(),
+            self.get_content_type().to_string().into(),
+        )];
+        let mut api_key = self.get_auth_header(&req.connector_auth_type)?;
+        header.append(&mut api_key);
+        Ok(header)
+    }
+}
+
+impl ConnectorCommon for Prophetpay {
+    fn id(&self) -> &'static str {
+        "prophetpay"
+    }
+
+    fn get_currency_unit(&self) -> api::CurrencyUnit {
+        api::CurrencyUnit::Minor
```

Comment:
This is base unit for this connector, please change this in the implementation PR.